client/blocks/app-banner/index.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import { getSectionName } from 'state/ui/selectors';
+import { getPreference } from 'state/preferences/selectors';
+import { isNotificationsOpen } from 'state/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { savePreference } from 'state/preferences/actions';
+import {
+	identity,
+	includes,
+	noop
+} from 'lodash';
+import {
+	ALLOWED_SECTIONS,
+	getAppBannerData,
+	getNewDismissTimes,
+	getCurrentSection,
+	isDismissed,
+	PREFERENCE_NAME,
+} from './utils';
+
+class AppBanner extends Component {
+	static propTypes = {
+		saveDismissTime: PropTypes.func,
+		translate: PropTypes.func,
+		trackAppBannerImpression: PropTypes.func,
+		trackAppBannerDismiss: PropTypes.func,
+		trackAppBannerOpen: PropTypes.func,
+		userAgent: PropTypes.string,
+		// connected
+		currentSection: React.PropTypes.string,
+		dismissedUntil: React.PropTypes.array,
+	};
+
+	static defaultProps = {
+		saveDismissTime: noop,
+		translate: identity,
+		trackAppBannerImpression: noop,
+		trackAppBannerDismiss: noop,
+		trackAppBannerOpen: noop,
+		userAgent: navigator.userAgent,
+	};
+
+	isVisible() {
+		const { dismissedUntil, currentSection } = this.props;
+
+		return this.isMobile() && ! isDismissed( dismissedUntil, currentSection );
+	}
+
+	isiOS() {
+		return this.props.userAgent.match( /iPhone/i ) ? true : false;
+	}
+
+	isAndroid() {
+		return this.props.userAgent.match( /Android/i ) ? true : false;
+	}
+
+	isMobile() {
+		return this.isiOS() || this.isAndroid();
+	}
+
+	dismiss = ( event ) => {
+		event.preventDefault();
+		const { currentSection } = this.props;
+
+		this.props.saveDismissTime( currentSection );
+		this.props.trackAppBannerDismiss( currentSection );
+	};
+
+	openApp = () => {
+		this.props.trackAppBannerOpen( this.props.currentSection );
+	};
+
+	getDeepLink() {
+		// TODO: update with real deep links when we get them
+		// just linking to respective app stores for now
+		if ( this.isAndroid() ) {
+			return 'https://play.google.com/store/apps/details?id=org.wordpress.android';
+		}
+
+		if ( this.isiOS() ) {
+			return 'https://itunes.apple.com/us/app/wordpress/id335703880';
+		}
+
+		return null;
+	}
+
+	render() {
+		const { translate, currentSection } = this.props;
+
+		if ( ! includes( ALLOWED_SECTIONS, currentSection ) ) {
+			return null;
+		}
+
+		if ( ! this.isVisible() ) {
+			return null;
+		}
```

Comment:
I'm aware of that, but since we need to toggle visibility if Notifications panel is open, I think we have to render it everywhere. We could move the same checks to parent too, but the performance gains seem marginal, and I'd like to keep App banner related code out of `layout/index.jsx` as much as possible.